### ISSUE TITLE
Predicting topic in a test dataset 

### ISSUE THREAD

**Issue by jihyunmd** [MaartenGr/BERTopic#1151]
Hi, thanks for developing BERTopic. I am really enjoying using it. 
I am currently working on predicting topics using trained models in different documents. But I get the exact same outputs when I run get.topic.info() and visualize_barchart() in the test set. 

My code is as below: 

from sklearn.model_selection import train_test_split

# train_test_split 
train_docs, test_docs = train_test_split(newdocs, test_size=0.8, random_state=42) 

# BERTopic

from bertopic import BERTopic
from umap import UMAP

##Prepare embeddings

from sentence_transformers import SentenceTransformer

sentence_model = SentenceTransformer("all-MiniLM-L6-v2")
embeddings = sentence_model.encode(train_docs, show_progress_bar=True)

##UMAP model
umap_model = UMAP(min_dist=0.0, metric='cosine', random_state=42)

# Train model
base_topic_model = BERTopic(umap_model=umap_model, verbose=True)
topics, probs = base_topic_model.fit_transform(train_docs, embeddings)

# Save Model
base_topic_model.save("pretrained_model80")

# Get topic info and visualize Barchart 
from IPython.display import clear_output
info=base_topic_model.get_topic_info()

import plotly.express as px
figpretrain=base_topic_model.visualize_barchart(custom_labels=True, top_n_topics=35)
figpretrain

################## Predict ##################

##Prepare embeddings
sentence_model = SentenceTransformer("all-MiniLM-L6-v2")
embeddings = sentence_model.encode(test_docs, show_progress_bar=True)

##Load saved instances
trained_topic_model = BERTopic.load("pretrained_model80")

##Extract new topics 
predicted_topics, predicted_probs = trained_topic_model.transform(test_docs, embeddings)

# Get predicted topic info and visualize Barchart 
infotest=trained_topic_model.get_topic_info()
infotest

figtest=trained_topic_model.visualize_barchart(custom_labels=True, top_n_topics=35)
figtest

After running this code, 
"info" and "infoest" / "figpretrain" and "figtest" are exactly the same. Was it supposed to be like this or did I do something wrong? 
Thank you!

**Comment by MaartenGr** [comment]
When you run `transform` it does not change the internal model but merely predicts what the topics are for `test_docs`. If you want to update it using unseen documents, you would either have to re-train the model or use [online topic modeling](https://maartengr.github.io/BERTopic/getting_started/online/online.html) instead.

**Comment by jihyunmd** [comment]
Thanks for the reply. I can understand that topic "name"s are the same in training and test set when I run .get_topic_info(), but aren't the "count"s suppose to be different in the test set? 
Thank you!

**Comment by MaartenGr** [comment]
> Thanks for the reply. I can understand that topic "name"s are the same in training and test set when I run .get_topic_info(), but aren't the "count"s suppose to be different in the test set?

No, traditionally, the `transform` function in many models, also outside of BERTopic like in scikit-learn, do not change anything from the original model.

**Comment by jihyunmd** [comment]
Thank you!
